### PR TITLE
Adds multi-trajectory support to offline mapping.

### PR DESCRIPTION
**Pull request by damonkohler**

### CODE REVIEW

File: cartographer_ros/cartographer_ros/offline_node_main.cc [hunk 1]
```diff
@@ -49,8 +49,18 @@ namespace {
 constexpr int kLatestOnlyPublisherQueueSize = 1;
 constexpr char kClockTopic[] = "clock";
 
-void Run() {
-  // TODO(damonkohler): Pull out this common code across binaries.
+std::vector<string> SplitString(const string& input, const char delimeter) {
```
Comment (wohe) [inline cartographer_ros/cartographer_ros/offline_node_main.cc]: typo: `delimiter`

File: cartographer_ros/cartographer_ros/offline_node_main.cc [hunk 2]
```diff
@@ -49,8 +49,18 @@ namespace {
...
+  std::stringstream stream;
```
Comment (wohe) [inline cartographer_ros/cartographer_ros/offline_node_main.cc]: `std::istringstream stream(input);`?
No need to call `str()`.

File: cartographer_ros/cartographer_ros/offline_node_main.cc [hunk 3]
```diff
@@ -106,61 +116,68 @@ void Run() {
...
 
-    rosgraph_msgs::Clock clock;
-    clock.clock = msg.getTime();
-    clock_publisher.publish(clock);
+    for (const rosbag::MessageInstance& msg : rosbag::View(bag)) {
+      if (!::ros::ok()) {
+        return;
+      }
+
+      const string& topic = node.node_handle()->resolveName(msg.getTopic());
```
Comment (wohe) [inline cartographer_ros/cartographer_ros/offline_node_main.cc]: `const string topic` to be explicit that `resolveName()` return a value?

File: cartographer_ros/cartographer_ros/offline_node_main.cc [hunk 4]
```diff
@@ -106,61 +116,68 @@ void Run() {
...
+      if (expected_sensor_ids.count(
+              node.node_handle()->resolveName(msg.getTopic())) == 0) {
```
Comment (wohe) [inline cartographer_ros/cartographer_ros/offline_node_main.cc]: Use `topic`?

File: cartographer_ros/cartographer_ros/offline_node_main.cc (hunk 1 above)
Comment (damonkohler) [inline cartographer_ros/cartographer_ros/offline_node_main.cc]: Done.

File: cartographer_ros/cartographer_ros/offline_node_main.cc (hunk 2 above)
Comment (damonkohler) [inline cartographer_ros/cartographer_ros/offline_node_main.cc]: Done.

**Pull request merged**